index.d.ts

```diff
@@ -0,0 +1,365 @@
+             */
+            name: string;
+            /**
+             * Contains async tests and their current completion status
+             */
+            async: any;
+            /**
+             * Overall errors count for this form
+             */
+            failCount: number;
+            /**
+             * Whether there are validation errors or not
+             */
+            hasValidationErrors: boolean;
+            /**
+             * Whether there are validation warnings or not
+             */
+            hasValidationWarnings: boolean;
+            /**
+             * All skipped fields (empty, unless the specific option is used)
+             */
+            skipped: any[];
+            /**
+             * Overall test count in this form
+             */
+            testCount: number;
+            /**
+             * Detailed stats per field (structure detailed below)
+             */
+            testsPerformed: any;
+            /**
+             * Actual errors per each field
+             */
+            validationErrors: any;
+            /**
+             * Actual errors per each field
+             */
+            validationWarnings: any;
+            /**
+             * Overall warnings count for this form
+             */
+            warnCount: number;
+            /**
+             * Getter function which allows accessing the errors array of a certain field (or the whole suite if not supplied)
+             */
+            getErrors: (field?: string) => any[];
+            /**
+             * Getter function which allows accessing the warnings array of a certain field (or the whole suite if not supplied)
+             */
+            getWarnings: (field?: string) => any[];
+            /**
+             * Returns whether a certain field (or the whole suite if not supplied) has errors
+             */
+            hasErrors: (field?: string) => boolean;
+            /**
+             * Returns whether a certain field (or the whole suite if not supplied) has warnings
+             */
+            hasWarnings: (field?: string) => boolean;
+        }
+
+        export type IFunctionOrPromise = () => void | Promise<any>;
+
+        export type IVERSION = string;
+        export type IWARN = 'warn';
+        export type IFAIL = 'fail';
+
+        export interface IValidate {
+            (): boolean;
+        }
+
+        export type IEnforceConstructor = {
+            new(): (value: any) => IEnforce;
+            new<T extends { [key: string]: (...args: any[]) => IEnforce<T> }>
+                (arg: T): (value: any) => IEnforce<T>;
+        };
+
+        export interface IEnforce<T = {}> {
+            /**
+             * Checks if a value contains a regex match by Regex expression
+             * 
+             * @example
+             *  enforce(1984).matches(/[0-9]/) // truthy
+             * 
+             *  enforce('ninety eighty four').matches(/[0-9]/) // falsy
```

Comment:
```suggestion
             *  enforce('nineteen eighty four').matches(/[0-9]/) // falsy
```
(I need to fix it in the docs too)